pkg/dockercreds/k8sdockercreds/k8s_keychain.go

```diff
@@ -76,10 +78,11 @@ func toStringPullSecrets(secrets []corev1.LocalObjectReference) []string {
 type annotatedBasicAuthKeychain struct {
 	secretRef     registry.SecretRef
 	secretFetcher *secret.Fetcher
+	ctx           context.Context
 }
 
 func (k *annotatedBasicAuthKeychain) Resolve(res authn.Resource) (authn.Authenticator, error) {
-	secrets, err := k.secretFetcher.SecretsForServiceAccount(k.secretRef.ServiceAccountOrDefault(), k.secretRef.Namespace)
+	secrets, err := k.secretFetcher.SecretsForServiceAccount(k.ctx, k.secretRef.ServiceAccountOrDefault(), k.secretRef.Namespace)
```

Comment:
I think this is exposing an awkward part of our design where we lazy load the secrets from the service account which is now requiring us to hold a context in the the struct.

Reply:
Eventually we probably should create a constructor for the `annotatedBasicAuthKeychain` to load the secrets like the `k8schain` does.

Reply:
Definitely should do this but, maybe outside the scope of the current PR. Should open new PR with this refactor.